core/commands/dag/dag.go

```diff
@@ -668,3 +671,134 @@ The output of blocks happens in strict DAG-traversal, first-seen, order.
 		},
 	},
 }
+
+type DagStat struct {
+	Size      uint64
+	NumBlocks int64
+}
+
+func (s *DagStat) String() string {
+	return fmt.Sprintf("Size: %d, NumBlocks: %d", s.Size, s.NumBlocks)
+}
+
+var DagStatCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "Gets stats for a DAG",
+		ShortDescription: `
+'ipfs dag size' fetches a dag and returns various statistics about the DAG.
+Statistics include size and number of blocks.
+
+Note: This command skips duplicate blocks in reporting both size and the number of blocks
```

Comment:
We should be able to report both sets of data (duplicates + not) together, but it requires a bit of state tracking to do it correctly. This didn't seem necessary for an initial release of this command, but I'm open to counter opinions.

@obo20 you had a request for this command would you prefer to see the data with the duplicate blocks or without?

Reply:
@aschmahmann I can see both ways having usefulness, but deduplicated would probably be the most useful since that's one of the strengths of IPFS.